### PR TITLE
chore(#324): upgrading to android 13

### PR DESCRIPTION
**Pull request by latin-panda**
**Description**

This is to upgrade from Android 11 to Android 13.
- READ_EXTERNAL_STORAGE is ignored. Therefore, we stopped requesting this permission for Android +13 with the prominent disclosure prompt. At the same time, we will keep asking this permission for any device in previous versions of Android (12, 11, 10, etc). 
  - All devices regardless of the Android version should be able to get files (png, video, audio, pdf, xls etc.) using [File Picker](https://github.com/medic/cht-android/issues/159) and [Android App Launcher](https://docs.communityhealthtoolkit.org/apps/reference/forms/app/#android-app-launcher)
- We limited the SDK target on the following tests because our current test dependencies don't support target 33
  - SmsSenderTest.java 
  - FilterableListAdapterTest.java
  - AsyncExecutorTest.java
  - VibratorTest.java

#324

### CODE REVIEW

File: src/main/java/org/medicmobile/webapp/mobile/RequestStoragePermissionActivity.java
Comment (latin-panda) [inline src/main/java/org/medicmobile/webapp/mobile/RequestStoragePermissionActivity.java:80]: Just in case if someone is wondering

> You can call finish() from within this function, in which case onDestroy() will be immediately called after onCreate(Bundle)

[source](https://developer.android.com/reference/android/app/Activity#onCreate(android.os.Bundle))

File: src/test/java/org/medicmobile/webapp/mobile/SmsSenderTest.java
Comment (latin-panda) [inline src/test/java/org/medicmobile/webapp/mobile/SmsSenderTest.java:49]: We're getting this error because our current version of `robolectric` doesn't support `target 33`

```
java.lang.IllegalArgumentException at RobolectricTestRunner.java:260
        Caused by: java.lang.IllegalArgumentException at DefaultSdkPicker.java:118
```
In this [link](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/docs/testing.md#robolectric), they recommend to use `Config(sdk = 31)` to make the test pass as a _temporary fix_.

**Comment by latin-panda** [comment]
@garethbowen, it turns out that our version of `robolectric` doesn't fully support target 33. 
I tried upgrading the library but it has deprecations that we need to fix and it requires upgrading `androidx` libraries, which also have more deprecations that we need to figure out. It's going to take longer to resolve them. 

Knowing that we have limited time before PlayStore blocks our apps, I've dropped the work on the tests and patch them by running them in target 31. I created a separate ticket to [fix the dependencies](https://github.com/medic/cht-android/issues/326).  Then @ngaruko and I can focus on testing.

My question is:

If the manual testing is good, can we release this to unblock the PlayStore and work on the dependencies later?

**Review by garethbowen — APPROVED** [review]

**Comment by garethbowen** [comment]
@latin-panda Yes let's go manual to unblock this - incomplete testing is better than blocking publishing altogether. But let's make sure we fix this soon - if you don't have time to take this on let me know and I'll shop it around.

Thanks for doing such a deep dive here!

**Review by ngaruko — COMMENTED** [review]
1. Test most critical [scenarios](https://github.com/medic/cht-android/issues/324#issuecomment-1692724352) and they all passed
2. Random testing of a few workflows
3. Reading from Google developer, one of the major changes around permissions is replacing `[READ_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE)` by [granular media permission](https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions). Apparently we removed the permission and don't seem to have introduced any granular permission. Is that intentional @latin-panda ?

**Comment by latin-panda** [comment]
> Reading from Google developer, one of the major changes around permissions is replacing [READ_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE) by [granular media permission](https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions). Apparently we removed the permission and don't seem to have introduced any granular permission. Is that intentional @latin-panda ?

@ngaruko I saw that, but I'm not clear under what circumstances it's needed; for example, they don't mention anything about the other types of files that aren't images, video, or audio, such as pdf or spreadsheets, and that we were accessing using `READ_EXTERNAL_STORAGE` (sample [form](https://github.com/medic/support-scripts/tree/master/forms/enketo_widgets)). 
Some blogs say that since it's in a shared place then it's not needed. When I tested it, it worked on my phone even when not requesting any of those permissions. 
- Are you experiencing any issue using the using https://github.com/medic/cht-android/issues/159 and [Android App Launcher](https://docs.communityhealthtoolkit.org/apps/reference/forms/app/#android-app-launcher) when accessing any type of file?
- How else can we test this and get feedback? Maybe we're missing they way to reproduce it.

In another topic: 
- Did you try the feature of sending SMS when submitting forms? I need help getting the form configured for that. There's a known [issue](https://github.com/medic/cht-android/issues/322), but we just don't want to make it worse

**Comment by garethbowen** [comment]
@latin-panda I don't have a lot of android experience either but I think your "reason 3" is spot on. Thanks for being so diligent!

**Comment by latin-panda** [comment]
@ngaruko waiting for your approval or feedback after you've done verifying the app

**Comment by ngaruko** [comment]
@latin-panda . Thanks for all the research. It all makes sense and practically, I do not see any scenario where access is denied due to the permission not present. So, good to go.

**Review by ngaruko — APPROVED** [review]
Good research on those permissions. 👍

**Pull request merged**
